src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1148,12 +1148,49 @@ impl Cage {
     }
 
     //------------------------------------STAT SYSCALL------------------------------------
+    /// ### Description
+    ///
+    /// `stat_syscall` retrieves file information for the file specified by
+    /// `path` and populates the provided `statbuf` with this information.
+    /// Although no file permissions are required to perform the call, execute
+    /// (search) permission is required on all of the directories in
+    /// pathname that lead to the file.
+    ///
+    /// ### Arguments
+    ///
+    /// It accepts two parameters:
+    /// * `path` - A string slice that specifies the file path for which status
+    ///   information is to be retrieved.
+    /// * `statbuf` - A mutable reference to a `StatData` struct where the file
+    ///   status will be stored.
+    ///
+    /// ### Returns
+    ///
+    /// For a successful call, the return value will be 0. On error, a negative
+    /// errno is returned to indicate the error.
+    ///
+    /// ### Errors
+    ///
+    /// * `ENOENT` - The file specified by `path` does not exist or the path is
+    ///   invalid.
+    ///
+    /// ### Panics
+    ///
+    /// * If the inode number retrieved from `metawalk` does not exist in
+    ///   `FS_METADATA.inodetable`.
+    /// * If `statbuf` is incorrectly populated, causing potential runtime
+    ///   issues.
+    ///
+    /// For more detailed description of all the commands and return values,
+    /// refer to the stat man page [here](https://man7.org/linux/man-pages/man2/stat.2.html).
 
     pub fn stat_syscall(&self, path: &str, statbuf: &mut StatData) -> i32 {
+        //convert the path to an absolute path of type `PathBuf`
         let truepath = normpath(convpath(path), self);
 
         //Walk the file tree to get inode from path
         if let Some(inodenum) = metawalk(truepath.as_path()) {
+            // can panic if inode does not exist in inode table
```

Comment:
I believe the case of nonexistent entry (which would have led to a panic) is checked in the conditional statement above. In other words, if there is no entry, metawalk would return None, and you won't get to the unwrap part.

Reply:
nice catch! kinda redundant that we are checking the table 2 times though, but it does improve code readability